src/sentry/sentry_metrics/kafka_metrics_interface.py

```diff
@@ -0,0 +1,145 @@
+        assert kafka_topic_dict is not None
+        cluster_name = kafka_topic_dict["cluster"]
+        producer_config = get_kafka_producer_cluster_options(cluster_name)
+        producer_config.pop("compression.type", None)
+        producer_config.pop("message.max.bytes", None)
+        self.producer: Producer = KafkaProducer(
+            build_kafka_configuration(default_config=producer_config)
+        )
+
+    def counter(
+        self,
+        use_case_id: UseCaseID,
+        org_id: int,
+        project_id: int,
+        metric_name: str,
+        value: Union[int, float],
+        tags: Mapping[str, str],
+        unit: Optional[str] = None,
+        retention_days: Optional[int] = None,
+    ) -> None:
+
+        """
+        Emit a counter metric for internal use cases only.
+        Ensure that the use_case_id passed in has been registered
+        in the UseCaseID enum.
+        """
+
+        counter_metric = {
+            "org_id": org_id,
+            "project_id": project_id,
+            "name": build_mri(metric_name, "c", use_case_id, unit),
+            "value": value,
+            "timestamp": int(datetime.now().timestamp()),
+            "tags": tags,
+            "retention_days": retention_days,
+            "type": "c",
+        }
+
+        payload = KafkaPayload(None, json.dumps(counter_metric).encode("utf-8"), [])
+        self.producer.produce(self.kafka_topic, payload)
+
+    def set(
+        self,
+        use_case_id: UseCaseID,
+        org_id: int,
+        project_id: int,
+        metric_name: str,
+        value: Sequence[int],
+        tags: Mapping[str, str],
+        unit: Optional[str] = None,
+        retention_days: Optional[int] = None,
+    ) -> None:
+
+        """
+        Emit a set metric for internal use cases only. Can support
+        a sequence of values. Ensure that the use_case_id passed in has
+        been registered in the UseCaseID enum.
+        """
+
+        set_metric = {
+            "org_id": org_id,
+            "project_id": project_id,
+            "name": build_mri(metric_name, "s", use_case_id, unit),
+            "value": value,
+            "timestamp": int(datetime.now().timestamp()),
+            "tags": tags,
+            "retention_days": retention_days,
+            "type": "s",
+        }
+
+        payload = KafkaPayload(None, json.dumps(set_metric).encode("utf-8"), [])
+        self.producer.produce(self.kafka_topic, payload)
+
+    def distribution(
+        self,
+        use_case_id: UseCaseID,
+        org_id: int,
+        project_id: int,
+        metric_name: str,
+        value: Sequence[Union[int, float]],
+        tags: Mapping[str, str],
+        unit: Optional[str] = None,
+        retention_days: Optional[int] = None,
+    ) -> None:
+
+        """
+        Emit a distribution metric for internal use cases only. Can
+        support a sequence of values. Ensure that the use_case_id passed in
+        has been registered in the UseCaseID enum.
```

Comment:
I don't think the second sentence is necessary. Given the `UseCaseID` enum type, it should be impossible to pass anything else anyway.

On the other hand, I would mention that this function immediately returns even if the value is not actually produced to the broker yet.